comms/dht/tests/dht.rs

```diff
@@ -536,6 +540,85 @@ async fn dht_propagate_dedup() {
     assert_eq!(count_messages_received(&received, &[&node_C_id]), 1);
 }
 
+#[tokio_macros::test]
+#[allow(non_snake_case)]
+async fn dht_repropagate() {
+    // Node C knows no one
+    let mut node_C = make_node(PeerFeatures::COMMUNICATION_NODE, []).await;
+    // Node B knows about Node C
+    let mut node_B = make_node(PeerFeatures::COMMUNICATION_NODE, [node_C.to_peer()]).await;
+    // Node A knows about Node B and C
+    let mut node_A = make_node(PeerFeatures::COMMUNICATION_NODE, [node_B.to_peer(), node_C.to_peer()]).await;
+    node_A.comms.peer_manager().add_peer(node_C.to_peer()).await.unwrap();
+    log::info!(
+        "NodeA = {}, NodeB = {}, Node C = {}",
+        node_A.node_identity().node_id().short_str(),
+        node_B.node_identity().node_id().short_str(),
+        node_C.node_identity().node_id().short_str(),
+    );
+
+    // Connect the peers that should be connected
+    async fn connect_nodes(node1: &mut TestNode, node2: &mut TestNode) {
+        node1
+            .comms
+            .connectivity()
+            .dial_peer(node2.node_identity().node_id().clone())
+            .await
+            .unwrap();
+    }
+    // Pre-connect nodes, this helps message passing be more deterministic
+    connect_nodes(&mut node_A, &mut node_B).await;
+    connect_nodes(&mut node_B, &mut node_C).await;
+    connect_nodes(&mut node_C, &mut node_A).await;
```

Comment:
Should this not rather read like this?
```suggestion
    connect_nodes(&mut node_A, &mut node_C).await;
```

Reply:
Can do that, doesn't really make a difference here.